service/src/main/java/uk/nhs/adaptors/gp2gp/common/storageconnectors/AzureStorageConnector.java

```diff
@@ -0,0 +1,54 @@
+package uk.nhs.adaptors.gp2gp.common.storageconnectors;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.commons.lang3.StringUtils;
+import com.azure.storage.blob.BlobClient;
+import com.azure.storage.blob.BlobContainerClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.BlobServiceClientBuilder;
+
+public class AzureStorageConnector implements StorageConnector {
+
+    private BlobServiceClient blobServiceClient;
+    private BlobContainerClient containerClient;
+
+    protected AzureStorageConnector(StorageConnectorConfiguration configuration) {
+        String connectionString = configuration.getAzureConnectionString();
```

Comment:
Constructors should be used only to assign fields. Could you define a bean in a configuration class (method with @Bean annotation) which returns a prepared "BlobContainerClient" and then simply inject it here in a constructor?